### DISCUSSION TITLE
Why is the 'photos' variant not in F-Droid

### DISCUSSION THREAD

**Discussion by verbumfeit** [lukaspieper/Gcam-Services-Provider#88] [General]
Hey there,

first: thank you for your work, I'm currently installing GrapheneOS on my new Pixel, I am very excited to make full use of my cameras!

Question: I read that the 'photos' variant of your app is not on F-Droid because of some guideline it violates. I'm just wondering what guideline that is and why this variant has to violate it to function properly?

Thank you for any imformation :)

**Comment by lukaspieper** [comment]
Hi,

the "photos" variant is using the application id of Google Photos (`com.google.android.apps.photos`) for technical reasons. This conflicts with the following part of the F-Droid policy:

> The software should use its own unique Android “[Application ID](https://developer.android.com/studio/build/configure-app-module)” based on a domain name under the developer’s control. For example, if the app is in a group called “foo” on [gitlab.com](https://gitlab.com/), then the domain name could be io.gitlab.foo. It should not be based on someone’s domain name, (e.g. com.google.foo).

https://f-droid.org/en/docs/Inclusion_Policy/

You can get both variants from Izzy's F-Droid repository (that takes the binaries from the release section here on GitHub).

**Reply by verbumfeit** [reply]
Thank you so much for the explanation!

And the 'technical reasons' mentioned are, that the Gcam app only opens the app that has the Google Photos application ID when you tap on the most recently taken picture, I would guess.

**Reply by lukaspieper** [reply]
Yes, Gcam sends a so-called intent to the app with that application id and my app then redirects that request (intent) to your default gallery app.
